### PR TITLE
Patch - adjust resolvconf variable check and insert 'else' option

### PR DESCRIPTION
**Pull request by edsonflavio**
test of resolvconf isn't check comments. I adjust this check for not create files infinitaly and insert else option at the end of file  for message show correctly when no changes is necessary.

### CODE REVIEW

File: nsfailover.sh [hunk 1]
```diff
@@ -148,6 +148,7 @@ search ${NS_SEARCH}"
 
 # Load current config (without comments)
 current="$(cat "${NS_FILE}" |egrep -v '^#')" || true
+resolvconf=$(echo "$resolvconf" | egrep -v '^#')
```
Comment (kvz) [inline nsfailover.sh:151]: This seems to remove only `# Do not edit. Updated by cron process!!"` which is useful to have in that file tho?

Comment (edsonflavio) [reply nsfailover.sh:151]: You are correct. This line only remove the comments, but without this the if was returning true ever time and execute copy of /etc/resolv.cont to .bak causing a possible DoS on host. My patch correct this.

Comment (edsonflavio) [reply nsfailover.sh:151]: A variable resolvconf and content of resolv.conf every are diferent If you not remove comments in both.

Comment (kvz) [reply nsfailover.sh:151]: what about adding a `commentLessResolvConf` var or similar, that we run the compare against, but that we do not write to disk?

Comment (edsonflavio) [reply nsfailover.sh:151]: The file write disk have all coment necessary,  but only have change on the file. Insert this var is a option.

Comment (edsonflavio) [reply nsfailover.sh]: Hi, I review my code here and adding a commentLessResolveConf for test with $current, resolve the problem.
I make this and are ok now.

File: nsfailover.sh [hunk 2]
```diff
@@ -39,16 +39,16 @@
 
 # Environment variables
 [ -z "${LOG_LEVEL}" ]       && LOG_LEVEL="6" # 7 = debug, 0 = emergency
-[ -z "${NS_ENABLE}" ]       && NS_ENABLE="no" # Set to no to disable
-[ -z "${NS_TESTDOMAIN}" ]   && NS_TESTDOMAIN="google.com" # Use this to determine if NS is healthy
-[ -z "${NS_1}" ]            && NS_1="" # Primary Nameserver (172.16.0.23 for Amazon EC2). You need to set this yourself
-[ -z "${NS_2}" ]            && NS_2="8.8.8.8" # Secundary Nameserver: Google
-[ -z "${NS_3}" ]            && NS_3="4.2.2.2" # Tertiary Nameserver: Level3
+[ -z "${NS_ENABLE}" ]       && NS_ENABLE="yes" # Set to no to disable
+[ -z "${NS_TESTDOMAIN}" ]   && NS_TESTDOMAIN="servicos.nc.ufpr.br" # Use this to determine if NS is healthy
```
Comment (kvz) [inline nsfailover.sh:43]: seems like some unrelated changes leaked into the pr

File: nsfailover.sh [hunk 3]
```diff
@@ -140,17 +140,20 @@ info "Best nameserver is ${use_level} (${use_server})"
...
 options timeout:${NS_TIMEOUT} attempts:${NS_ATTEMPTS}"
 # Optionally add search parameter
 [ -n "${NS_SEARCH}" ] && resolvconf="${resolvconf}
 search ${NS_SEARCH}"
 
 # Load current config (without comments)
-current="$(cat "${NS_FILE}" |egrep -v '^#')" || true
+current="$(cat "${NS_FILE}" | egrep -v '^#')" || true
+
+LessCommentResolv=$(echo "$resolvconf" | egrep -v '^#')
```
Comment (kvz) [inline nsfailover.sh:153]:
```suggestion
resolvNoComment=$(echo "$resolvconf" | egrep -v '^#')
```

File: nsfailover.sh [hunk 4]
```diff
@@ -140,17 +140,20 @@ info "Best nameserver is ${use_level} (${use_server})"
...
 
 # Is the config updated?
-if [ "${resolvconf}" != "${current}" ]; then
+if [ "${LessCommentResolv}" != "${current}" ]; then
```
Comment (kvz) [inline nsfailover.sh:156]:
```suggestion
if [ "${resolvNoComment}" != "${current}" ]; then
```

File: nsfailover.sh (hunk 3 above)
Comment (edsonflavio) [reply nsfailover.sh]: I create a new commit to resolv this question.

File: nsfailover.sh (hunk 4 above)
Comment (edsonflavio) [reply nsfailover.sh]: I create a new commit to resolv this question.

File: nsfailover.sh (hunk 2 above)
Comment (edsonflavio) [reply nsfailover.sh]: Yes. Clean in commit 
805f8b0

**Review by kvz — APPROVED** [review]
Thanks!
